analyzer/windows/tests/test_analyzer.py

```diff
@@ -684,3 +692,351 @@ def test_choose_package_zip_compound(self):
         self.assertEqual("modules.packages.zip_compound", pkg_name)
         self.assertEqual(pkg_class.__class__.__name__, "ZipCompound")
 
+
+class TestAnalyzerMonitoring(unittest.TestCase):
+    def setUp(self):
+        self.patches = []
+        patch_sleep = patch("lib.common.defines.KERNEL32.Sleep")
+        _ = patch_sleep.start()
+        self.addCleanup(patch_sleep.stop)
+        patch_call = patch("subprocess.call")
+        self.call = patch_call.start()
+        self.addCleanup(patch_call.stop)
+        self.analyzer = Analyzer()
+        self.cph = CommandPipeHandler(self.analyzer)
+        # Since the CommandPipeHandler.ignore_list is a class variable,
+        # we reset it between tests, so we get a fresh start.
+        self.cph.ignore_list = dict(pid=[])
+
+    def test_can_instantiate(self):
+        self.assertIsInstance(self.analyzer, Analyzer)
+        self.assertIsInstance(self.cph, CommandPipeHandler)
+
+    def test_handle_loaded(self):
+        random_pid = random.randint(1, 99999999)
+        with patch("analyzer.INJECT_LIST", [random_pid]):
+            ana = self.analyzer
```

Comment:
this would work the same, right?

```suggestion
        ana = self.analyzer
        with patch("analyzer.INJECT_LIST", [random_pid]):
```

Where it's possible, I'd suggest limiting the patched context. So if `ana = self.analyzer` doesn't need the patches, pull it out. This will help us in the future as we improve the analyzer more.